lib/reek/smells/smell_repository.rb

```diff
@@ -43,25 +42,21 @@ def examine(scope)
       end
 
       def detectors
-        @initialized_detectors ||= begin
-          @detectors = {}
-          smell_types.each do |klass|
-            @detectors[klass] = klass.new(source_via)
-          end
-          @detectors
-        end
+        @initialized_detectors ||= smell_types.map do |klass|
+          { klass => klass.new(source_via) }
+        end.reduce({}, :merge)
       end
 
-      private
+      protected
```

Comment:
> Then how about switching back to that gem you proposed to introduce before?

Given that I can see the point of private readers, I'd say yes.

> SmellRepository does not have any children. Making this protected is highly misleading then since I'd naturally assume it does.

That's not why it's misleading: Given Ruby's semantics, it suggests SmellRepository instances need to access each other's configuration and other values.